docs/integrations/data-ingestion/azure-data-factory/index.md

```diff
@@ -0,0 +1,503 @@
+Dataset.
+
+1. Download a [sample file](https://clickhouse-public-datasets.s3.eu-central-1.amazonaws.com/sensors/monthly/2019-06_bmp180.csv.zst)
+   from the [Environmental Sensors Dataset](https://clickhouse.com/docs/getting-started/example-datasets/environmental-sensors)
+
+2. In the Azure Portal, create a new storage account if you don't already have one.
+
+:::warning
+Make sure that **Allow storage account key access** is enabled for your storage
+account, otherwise you will not be able to use the account keys to access the
+data.
+:::
+
+3. Create a new container in your storage account. In this example, we name it sensors.
+   You can skip this step if you're using an existing container.
+
+4. Upload the previously downloaded `2019-06_bmp180.csv.zst` file to the
+   container.
+
+5. Follow the steps described earlier to obtain the Azure Blob Storage
+   connection string.
+
+Now that everything is set up, you can query the data directly from Azure Blob Storage:
+
+    ```sql
+    SELECT *
+    FROM azureBlobStorage(
+        '<YOUR CONNECTION STRING>', 
+        'sensors',
+        '2019-06_bmp180.csv.zst', 
+        'CSVWithNames')
+    LIMIT 10
+    SETTINGS format_csv_delimiter = ';'
+    ```
+
+7. To load the data into a table, create a simplified version of the
+   schema used in the original dataset:
+    ```sql
+    CREATE TABLE sensors
+    (
+        sensor_id UInt16,
+        lat Float32,
+        lon Float32,
+        timestamp DateTime,
+        temperature Float32
+    )
+    ENGINE = MergeTree
+    ORDER BY (timestamp, sensor_id);
+    ```
+
+:::info
+For more information on configuration options and schema inference when
+querying external sources like Azure Blob Storage, see [Automatic schema
+inference from input data](https://clickhouse.com/docs/interfaces/schema-inference)
+:::
+
+8. Now insert the data from Azure Blob Storage into the sensors table:
+    ```sql
+    INSERT INTO sensors
+    SELECT sensor_id, lat, lon, timestamp, temperature
+    FROM azureBlobStorage(
+        '<YOUR CONNECTION STRING>', 
+        'sensors',
+        '2019-06_bmp180.csv.zst', 
+        'CSVWithNames')
+    SETTINGS format_csv_delimiter = ';'
+    ```
+
+Your sensors table is now populated with data from the `2019-06_bmp180.csv.zst`
+file stored in Azure Blob Storage.
+
+### Additional Resources {#additional-resources}
+
+This is just a basic introduction to using the azureBlobStorage function. For
+more advanced options and configuration details, please refer to the official
+documentation:
+
+- [azureBlobStorage Table Function](https://clickhouse.com/docs/sql-reference/table-functions/azureBlobStorage)
+- [Formats for Input and Output Data](https://clickhouse.com/docs/sql-reference/formats)
+- [Automatic schema inference from input data](https://clickhouse.com/docs/interfaces/schema-inference)
+
+## Using ClickHouse HTTP Interface in Azure Data Factory {#using-clickhouse-http-interface-in-azure-data-factory}
```

Comment:
This needs to be two separate documents @slabko one for azureBlobStorage table function and one for the push approach